File: gitdot-web/packages/api/src/endpoint/webhook/list-webhooks.ts
import { z } from "zod";
import { WebhookResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListWebhooksResponse = z.array(WebhookResource);
export type ListWebhooksResponse = z.infer<typeof ListWebhooksResponse>;

export const ListWebhooks = {
  path: "/repository/{owner}/{repo}/webhooks",
  method: "GET",
  request: z.object({}),
  response: ListWebhooksResponse,
} as const satisfies Endpoint;
export type ListWebhooks = typeof ListWebhooks;

File: gitdot-api/src/endpoint/webhook/list_webhooks.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{WebhookResource, common::Page},
};

pub struct ListWebhooks;

impl Endpoint for ListWebhooks {
    const PATH: &'static str = "/repository/{owner}/{repo}/webhooks";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListWebhooksRequest;
    type Response = ListWebhooksResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListWebhooksRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListWebhooksResponse = Page<WebhookResource>;
